go-controller/cmd/ovnkube/ovnkube.go

```diff
@@ -472,34 +481,29 @@ func runOvnKube(ctx context.Context, runMode *ovnkubeRunMode, ovnClientset *util
 			return fmt.Errorf("error when trying to initialize libovsdb SB client: %v", err)
 		}
 
-		cm, err := controllerManager.NewNetworkControllerManager(ovnClientset, runMode.identity,
-			masterWatchFactory, libovsdbOvnNBClient, libovsdbOvnSBClient, eventRecorder, wg)
+		cm, err = controllerManager.NewNetworkControllerManager(ovnClientset, masterWatchFactory, libovsdbOvnNBClient, libovsdbOvnSBClient, eventRecorder, wg)
 		if err != nil {
 			return err
 		}
-		err = cm.Start(ctx)
-		if err != nil {
-			return fmt.Errorf("failed to start ovnkube controller: %w", err)
-		}
-		defer cm.Stop()
 
-		// record delay until ready
-		metrics.MetricOVNKubeControllerReadyDuration.Set(time.Since(startTime).Seconds())
+		// start NetworkControllerManager in a separate goroutine to allow parallel startup for NodeNetworkControllerManager.
+		// This is only required in ovnkube-controller + ovnkube-node mode, but shouldn't hurt other configs
+		go func() {
+			err := cm.Start(ctx)
+			if err != nil {
+				ovnkubeControllerStartChan <- fmt.Errorf("failed to start ovnkube controller: %w", err)
+			}
+			// record delay until ready
+			metrics.MetricOVNKubeControllerReadyDuration.Set(time.Since(startTime).Seconds())
+			ovnkubeControllerStartChan <- nil
```

Comment:
Wouldn't the standard thing to do here be closing the channel?

Reply:
You made me google :D 
I couldn't find a common way to do so, looks like it is more of a personal preference, e.g. here is a common example, where a value returned by another function is passed to the channel https://go.dev/blog/pipelines "Parallel digestion" section. I find it more clear to send an error as if it was returned by another function, so explicitly say "error is nil" instead of "no error will be sent", but that is very subjective, so I am happy to change that is you find closing a better option :)